spec/services/authentication/omniauth_service_spec.rb

```diff
@@ -98,14 +98,15 @@
 
       it 'does not call register user service and logs in the user' do
         # should update the user attributes
-        expect(user).to receive(:update).with(user_attributes)
+        allow(user).to receive(:save)
```

Comment:
@opf/backend That is a cop that restricts the use of `expect(..).to receive` with the following pattern:

```ruby
allow(...).to receive

...

expect(...).to have_received
```

The reasoning behind it is the four phase test https://thoughtbot.com/blog/four-phase-test with expectations at the bottom of the spec. Let me know what you think, I'd probably ignore that cop and let users decide their testing style

Reply:
I think this is a good idea. It may not always be applicable, for instance when you want to test sends to modules or in general objects you don't want to or can't be replaced by a spy object.

Reply:
@oliverguenther That being said it would still be nice if the test would somehow confirm that the user attributes were updated correctly which does not happen anymore after this change.

Reply:
Thanks, fixed